src/gui/qgsmaptoolcapture.cpp

```diff
@@ -289,10 +291,40 @@ bool QgsMapToolCapture::tracingAddVertex( const QgsPointXY &point )
     mSnappingMatches.append( QgsPointLocator::Match() );
   }
 
+  // If the layer supports curves, we de-approximate curves
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mCanvas->currentLayer() );
+  if ( vlayer->dataProvider()->capabilities().testFlag( QgsVectorDataProvider::Capability::CircularGeometries ) )
+  {
+    // We convert the capture curve to OGR geometry
+    OGRGeometry *ogrCaptureCurve;
+    OGRGeometryFactory::createFromWkt( mCaptureCurve.asWkt().toLocal8Bit().data(), nullptr, &ogrCaptureCurve );
```

Comment:
we shouldn't use wkt as the inbetween here -- wkb should be used instead. And there's already QgsOgrUtils::ogrGeometryToQgsGeometry which is an optimised method to convert back from OGR to QGIS.

Reply:
Ok (done)